src/actions/__tests__/addAllMulticursor.ts

```diff
@@ -0,0 +1,119 @@
+import State from '../../@types/State'
+import addAllMulticursor from '../../actions/addAllMulticursor'
+import addMulticursor from '../../actions/addMulticursor'
+import newSubthought from '../../actions/newSubthought'
+import newThought from '../../actions/newThought'
+import contextToPath from '../../selectors/contextToPath'
+import setCursor from '../../test-helpers/setCursorFirstMatch'
+import hashPath from '../../util/hashPath'
+import initialState from '../../util/initialState'
+import reducerFlow from '../../util/reducerFlow'
+
+describe('addAllMulticursor', () => {
+  it('adds all siblings to multicursor when cursor is set', () => {
+    const steps = [newThought('a'), newThought('b'), newThought('c'), setCursor(['b']), addAllMulticursor]
+
+    const stateNew = reducerFlow(steps)(initialState())
+
+    const a = contextToPath(stateNew, ['a'])!
+    const b = contextToPath(stateNew, ['b'])!
+    const c = contextToPath(stateNew, ['c'])!
+
+    expect(stateNew.multicursors).toEqual({
+      [hashPath(a)]: a,
+      [hashPath(b)]: b,
+      [hashPath(c)]: c,
+    })
+    expect(stateNew.cursorBeforeMulticursor).toEqual(b)
+  })
+
+  it('adds all root thoughts to multicursor when no cursor is set', () => {
+    const steps = [newThought('a'), newThought('b'), newThought('c'), setCursor(null), addAllMulticursor]
+
+    const stateNew = reducerFlow(steps)(initialState())
+
+    const a = contextToPath(stateNew, ['a'])!
+    const b = contextToPath(stateNew, ['b'])!
+    const c = contextToPath(stateNew, ['c'])!
+
+    expect(stateNew.multicursors).toEqual({
+      [hashPath(a)]: a,
+      [hashPath(b)]: b,
+      [hashPath(c)]: c,
+    })
+    expect(stateNew.cursorBeforeMulticursor).toBeNull()
+  })
+
+  it('does not add subthoughts to multicursor', () => {
+    const steps = [
+      newThought('a'),
+      newSubthought('a1'),
+      setCursor(['a']),
+      newThought('b'),
+      setCursor(['a']),
+      addAllMulticursor,
+    ]
+
+    const stateNew = reducerFlow(steps)(initialState())
+
+    const a = contextToPath(stateNew, ['a'])!
+    const b = contextToPath(stateNew, ['b'])!
+
+    expect(stateNew.multicursors).toEqual({
+      [hashPath(a)]: a,
+      [hashPath(b)]: b,
+    })
+    expect(stateNew.cursorBeforeMulticursor).toEqual(a)
+  })
+
+  it('works with existing multicursors', () => {
+    const steps = [
+      newThought('a'),
+      newThought('b'),
+      newThought('c'),
+      newThought('d'),
+      setCursor(['b']),
+      (state: State) => addMulticursor(state, { path: contextToPath(state, ['c'])! }),
```

Comment:
We can use a little currying magic to get the simpler syntax that `newThought` and other action-creators use.

Replace `export default addMulticursor` with `export default _.curryRight(addMulticursor)`.

Then you can call `addMulticursor` with just its payload and get back a unary action-creator that the latest state will pass through:

```
const steps = [
      newThought('a'),
      newThought('b'),
      newThought('c'),
      newThought('d'),
      setCursor(['b']),
      addMulticursor({ path: contextToPath(state, ['c'])! }),
      addAllMulticursor,
]

Reply:
Don't think this'll work since I need to get 'c' from the new state after inserting the thoughts.

Reply:
Oops! You're right. 

The `addAllMulticursor` definition should be curried anyway to be consistent with the other action-creators.